Fix ExactTarget_* class autoloading

drsnyder opened the pull request:
The celery tasks for the `ExactTargetUpdate` extension are spewing errors like the following regarding `ExactTarget_RetrieveRequest` (but not limited to that class):

```
PHP Fatal Error: Class 'ExactTarget_RetrieveRequest' not found in /usr/wikia/slot1/9585/src/extensions/wikia/ExactTargetUpdates/api/ExactTargetApiHelper.php on line 64
```

This is because the `ExactTarget_*` classes (about 200) are declared in [exacttarget_soap_client.php](https://github.com/Wikia/app/blob/dev/extensions/wikia/ExactTargetUpdates/lib/exacttarget_soap_client.php) along with the `ExactTargetSoapClient` class. The autoloading in the setup class only references [`ExactTargetSoapClient`](https://github.com/Wikia/app/blob/dev/extensions/wikia/ExactTargetUpdates/ExactTargetUpdates.setup.php#L78) which means you have to reference `ExactTargetSoapClient` (e.g. by creating a new object) before you can reference any of the `ExactTarget_*`. This creates a load order dependency that results in class not found exceptions if it's not followed.

This PR addresses the issue by moving all of the `ExactTarget_*` classes into their own files one per class. The class loading in the setup file is also adjusted to include each class separately. Finally a test was added to confirm the loading works as expected.

Finally, I removed the `testDeleteSubscriber` test. It was testing the mock and was not confirming the expected result.

https://wikia-inc.atlassian.net/browse/SUS-129
https://wikia-inc.atlassian.net/browse/SUS-128

@Wikia/sustaining-team 

TODO:
- [x] Fix the tests
- [x] Add tests with a little more substance one or more classes that depend on the `ExactTarget_*` classes

Commit:
Create a failing test for ExactTargetApiHelper

The wrapRetriveRequest method fails with "Class
'ExactTarget_RetrieveRequest' not found". Add a test so we can fix it.

Commit:
Break out the ExactTarget_* classes into files

There should only be one class per file.

Commit:
Remove testDeleteSubscriber; it was testing the mock

Remove the testDeleteSubscriber test. It was testing the mock.

Commit:
Fix ExactTargetApiHelperTest name

kvas-damian commented on extensions/wikia/ExactTargetUpdates/tests/ExactTargetApiTest.php:
```diff
@@ -60,30 +60,6 @@ function testPrepareSoapVarsShouldReturnSoapVarsArray() {
-		$mockApiSubscriber->deleteRequest($aApiParams);
-	}
-
-	/**
```
This test was in fact testing `prepareSubscriberObjects` method of `\Wikia\ExactTarget\ExactTargetApiHelper`. What do you think about adding unit tests for that method?

drsnyder replied:
Using a mock of the class you want to test is problematic. Its not a good practice and usually indicates there is an issue with the design.

I think it would be better to test `prepareSubscriberObjects` directly. I'll look at implementing that.

kvas-damian replied:
Direct test for `prepareSubscriberObjects` is exactly what tried to say ;)

kvas-damian commented:
@drsnyder I believe you used some automated tool to extract those classes. It's not possible for a human to catch errors there ;)

drsnyder commented:
Yeah, that's too much to do by hand. It went something like this

``` perl
while ($string =~ /(?:^|\s*)(class ExactTarget_\w+ {[^}]*}$)/mpg) {
 # take out the name of the class
 # write the class file with $string & close
}
```

Commit:
Add test for prepareSubscriberObjects

Commit:
Remove the note about loading ExactTargetSoapClient first

Removed the note about loading ExactTargetSoapClient first. It's no
longer necessary since each class can be loaded individually as
referenced.

Commit:
Add a test for wrapCreateRequest

Commit:
add a test for wrapSimpleFilterPart
